swarm/network/protocol_test.go

```diff
@@ -225,16 +237,59 @@ func TestBzzHandshakeVersionMismatch(t *testing.T) {
 }
 
 func TestBzzHandshakeSuccess(t *testing.T) {
+	lightNode := false
 	addr := RandomAddr()
-	s := newBzzHandshakeTester(t, 1, addr)
+	s := newBzzHandshakeTester(t, 1, addr, lightNode)
 	id := s.IDs[0]
 
 	err := s.testHandshake(
-		correctBzzHandshake(addr),
-		&HandshakeMsg{Version: 5, NetworkID: 3, Addr: NewAddrFromNodeID(id)},
+		correctBzzHandshake(addr, lightNode),
+		&HandshakeMsg{Version: DefaultVersion, NetworkID: DefaultNetworkID, Addr: NewAddrFromNodeID(id)},
+	)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestBzzHandshakeLightNodeOff(t *testing.T) {
+	randomAddr := RandomAddr()
+	pt := newBzzHandshakeTester(t, 1, randomAddr, false)
+	id := pt.IDs[0]
+	addr := NewAddrFromNodeID(id)
+	peerLightNode := false
+
+	err := pt.testHandshake(
+		correctBzzHandshake(randomAddr, false),
+		&HandshakeMsg{Version: DefaultVersion, NetworkID: DefaultNetworkID, Addr: addr, LightNode: peerLightNode},
 	)
 
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	if pt.bzz.handshakes[id].LightNode != peerLightNode {
+		t.Fatal(fmt.Sprintf("peer LightNode flag is %v, should be %v", pt.bzz.handshakes[id].LightNode, peerLightNode))
+	}
+}
+
+func TestBzzHandshakeLightNodeOn(t *testing.T) {
```

Comment:
What's the point of those 2 tests? I think you only need one of them.

Reply:
The only reason I did two tests were to make sure it is set correctly independently of the default value. If we only check the On case (where `LightNode` is `True`), then it may happen that in the code it's set to True incorrectly buy tests wouldn't catch that.

I don't have a strong opinion about it but generally I think it doesn't hurt to have an extra test and it may cover errors as described above.